xtrack/beam_elements/elements_src/electroncooler.h

```diff
@@ -0,0 +1,124 @@
+// copyright ############################### //
+// This file is part of the Xtrack Package.  //
+// Copyright (c) CERN, 2021.                 //
+// ######################################### //
+#define POW2(X) ((X)*(X))
+#define POW3(X) ((X)*(X)*(X))
+#define POW4(X) ((X)*(X)*(X)*(X))
+#ifndef XTRACK_ELECTRONCOOLER_H
+#define XTRACK_ELECTRONCOOLER_H
+
+/*gpufun*/
+void ElectronCooler_track_local_particle(ElectronCoolerData el, LocalParticle* part0){
```

Comment:
If you want to save the force you can use the element internal record.
See: https://xsuite.readthedocs.io/en/latest/dev_guide_xtbe_internal_record.html